internal/run/source.go

```diff
@@ -265,17 +261,28 @@ func (w *Workflow) RunSource(ctx context.Context, parentStep *workflowTracking.W
 		})
 	}
 
-	if !w.SkipLinting {
+	if w.SkipLinting {
+		w.OnSourceResult(sourceRes)
```

Comment:
This seems strange. Why not just call `w.OnSourceResult(sourceRes)` after the else block?